libs/wingc/src/jsify.rs

```diff
@@ -591,6 +591,30 @@ impl<'a> JSifier<'a> {
 			}
 			StmtKind::Break => CodeMaker::one_line("break;"),
 			StmtKind::Continue => CodeMaker::one_line("continue;"),
+			StmtKind::IfLet {
+				condition,
+				statements,
+				var_name,
+				else_statements,
+			} => {
+				let mut code = CodeMaker::default();
+				code.open(format!("if ({}) {{", self.jsify_expression(condition, ctx)));
```

Comment:
I think we need to check explicitly against null here. Consider this example:

```js
// wing
let x: bool? = false;
if let y = x {
  assert(y == false);
}
```

The current code would jsify it as
```js
// javascript
const x = false;
if (x) {
  const y = x;
  // assert code
}
```

which would result in the code in the branch not being executed since "false" is [falsey](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) in JavaScript. Instead we might need something like:

```js
// javascript
const x = false;
if (x != null) {
  const y = x;
  // assert code
}
```

Reply:
Oh this is a great catch!